pkg/controller/pxc/version.go

```diff
@@ -0,0 +1,98 @@
+package pxc
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+
+	api "github.com/percona/percona-xtradb-cluster-operator/pkg/apis/pxc/v1"
+	"github.com/percona/percona-xtradb-cluster-operator/pkg/pxc/app/statefulset"
+	"github.com/robfig/cron"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const jobName = "ensure-version"
+
+func (r *ReconcilePerconaXtraDBCluster) deleteEnsureVersion(id int) {
+	r.crons.crons.Remove(cron.EntryID(id))
+	delete(r.crons.jobs, jobName)
+}
+
+func (r *ReconcilePerconaXtraDBCluster) ensurePXCVersion(cr *api.PerconaXtraDBCluster, vs VersionService, sfs *statefulset.Node) error {
```

Comment:
should be executed only if `updateStrategy=SmartUpdate`

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/416/commits/b2e7d5971c0b528e2296a98b0941f415acc6436d